apollo-runtime/src/main/java/com/apollographql/apollo/cache/normalized/OptimisticNormalizedCache.java

```diff
@@ -128,9 +130,9 @@
     return changedCacheKeys;
   }
 
-  private final class RecordJournal {
+  private static final class RecordJournal {
     Record snapshot;
-    final LinkedList<Record> history = new LinkedList<>();
+    final List<Record> history = new ArrayList<>();
```

Comment:
Is it intentionally?

Reply:
Yes, this is apart of `JdkObsolete`  from error prone. `ArrayList` and `ArrayDeque` are more performant.